dataplane/netlink.cpp

```diff
@@ -0,0 +1,261 @@
+#include "netlink.hpp"
+
+#include <variant>
+
+#include <net/if.h>
+#include <netlink/netlink.h>
+#include <netlink/route/neighbour.h>
+
+namespace netlink
+{
+
+std::variant<Entry, int> ParseNeighbor(
+        rtnl_neigh* neigh,
+        std::function<std::optional<tInterfaceId>(const char*)> get_id)
+{
+	int sysifid = rtnl_neigh_get_ifindex(neigh);
+	Entry entry;
+	char ifname[IFNAMSIZ];
+	if (if_indextoname(sysifid, ifname) == nullptr)
+	{
+		YANET_LOG_INFO("Skipping message for unknown OS interface '%i'\n", sysifid);
+		return NL_OK;
+	}
+	if (auto id = get_id(ifname); id.has_value())
+	{
+		std::get<tInterfaceId>(entry) = id.value();
```

Comment:
using a structure will avoid those `std::get`'s that are all over the place right now